src/corva/api.py

```diff
@@ -126,3 +128,59 @@ def _validate_timeout(self, timeout: int) -> bool:
             raise ValueError(
                 f'Timeout must be between {self.TIMEOUT_LIMITS[0]} and {self.TIMEOUT_LIMITS[1]} seconds.'
             )
+
+    def get_dataset(
+        self,
+        provider: str,
+        dataset: str,
+        query: dict,
+        sort: dict,
+        limit: int,
+        fields: Optional[str] = None,
+    ) -> List[dict]:
+        """Fetches all data from the endpoint '/api/v1/data/{provider}/{dataset}/'.
+
+        Args:
+          provider: company name, that owns the dataset.
+          dataset: dataset name.
+          query: search conditions. Example: {"asset_id": 123} - will fetch data
+            for asset with id 123.
+          sort: sort conditions. Example: {"timestamp": 1} - will sort data
+            in ascending order by timestamp.
+          limit: number of data points to fecth in one request. Example: if limit is 2
+            and there are 3 data points in the dataset - 2 api requests will be made.
+            Recommendation for setting this value:
+              1. The bigger ↑ each data point is - the smaller ↓ the limit;
+              2. The smaller ↓ each data point is - the bigger ↑ the limit.
+          fields: comma separated list of fields to return. Example: "_id,data".
+
+        Raises:
+          requests.HTTPError: if some request was unsuccessful.
+
+        Returns:
+          all data from dataset.
+        """
+
+        result = []
+
+        for skip in itertools.count(0, limit):
```

Comment:
We should not do this to prevent users from killing API/Mongo. For example, WITS collection has millions of records per asset. Just make user to pass the limit, max is 10 000. User can make the loop if more is needed.

Reply:
What do you think about using generator, so that users can loop over data easier, like following:

```python3
def get_dataset(
    self,
    provider: str,
    dataset: str,
    query: dict,
    sort: dict,
    limit: int,
    skip: int = 0,
    fields: Optional[str] = None,
) -> Iterator[List[dict]]:

    for loop_skip in itertools.count(skip, limit):
        response = self.get(
            f"/api/v1/data/{provider}/{dataset}/",
            params={
                "query": json.dumps(query),
                "sort": json.dumps(sort),
                "fields": fields,
                "limit": limit,
                "skip": loop_skip,
            },
        )
        response.raise_for_status()

        new_data = list(response.json())

        yield new_data

        if len(new_data) != limit:
            break
```

Usage example:

```python3
gen = api.get_dataset(
    provider="corva",
    dataset="dataset",
    query={"asset_id": 123},
    sort={"timestamp": 1},
    limit=1000,
)

# get a chunk of data
data = next(gen)

# get all data
all_data = list(gen)
```

Reply:
It's really, really rare occasion when you want to fetch more than 10 000 records.

Reply:
Thanks, reworked the function